pkg/runtime/handlerTable.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package runtime
+
+import (
+	"bytes"
+	"crypto/sha1"
+	"encoding/gob"
+	"sort"
+
+	"github.com/golang/glog"
+
+	"istio.io/mixer/pkg/adapter"
+	cpb "istio.io/mixer/pkg/config/proto"
+)
+
+type buildHandlerFn func(*cpb.Handler, []*cpb.Instance) (adapter.Handler, error)
+
+// HandlerEntry is an entry in the runtime handler table.
+// The entire handler table is initialized at startup and sha
+// is recorded per entry, so on config change the correct
+// handlerEntries can be replaced.
+type HandlerEntry struct {
+	// Name of the handler
+	Name string
+
+	// Handler is the initialized handler object.
+	Handler adapter.Handler
+
+	// HandlerCreateError records error while creating the handler
+	HandlerCreateError error
+
+	// Instances is the global list of instances associated with this handler
+	Instances map[string]bool
+
+	// sha is used to verify and update the handlerEntry.
+	// sha must use handler.Params since any change to Params
+	// requires creation of a new handler instance.
+	// Some handlers need to be told about data they are about to receive.
+	// For example prometheus adapter initialized objects based on this configuration.
+	// If new configInstances are associated with such a handler, it also necessitates
+	// recreation of the handler.
+	sha [sha1.Size]byte
+
+	// If this is set, the handler is closed.
+	notInUse bool
```

Comment:
This where we record that **this** entry needs to be cleaned up by `close()` ing the handler.